design/0005-simplify-state-management.md

```diff
@@ -0,0 +1,202 @@
+# Simplifying state management in Linkerd
+
+- Contribution Name: Simplify State Management
+- Implementation Owner: @Pothulapati
+- Start Date:
+- Target Date: 2.9 release
+- RFC PR: [linkerd/rfc#0000](https://github.com/linkerd/rfc/pull/0000)
+- Linkerd Issue:
+  [linkerd/linkerd2#0000](https://github.com/linkerd/linkerd2/issues/0000)
+- Reviewers: @adleong @alpeb
+
+## Summary
+
+[summary]: #summary
+
+Currently, Linkerd state in the `linkerd-config` ConfigMap as flags parsed from `installOptions`.
+This has become counter intuitive after we added Helm support as `Values` struct
+became our main source of truth (for rendering, etc), and causing back and forth
+conversions between `installOptions` and `Values`. This RFC aims to remove the
+unnecessary complexity in managing state and simplify things.
```

Comment:
I think it's worth framing the problem at first a little less technically, clarifying that the goal is to simplify state management for _upgrades_